packages/frontapp/src/app/pages/HistoryOfDonatePage.tsx

```diff
@@ -1,7 +1,41 @@
 import React from 'react';
+import { TextArticle } from '../globalUI/TextArticle';
+import { TextBase } from '../globalUI/TextBase';
+import BookCard from '../components/BookCard';
+import { List, Wrapper } from './HistoryOfClaimPage';
+import { useTranslation } from 'react-i18next';
+import { useAppSelector } from '../hooks/useTypedSelector';
+import { useGetAllDonatedMaterialsByPersonQuery } from '@mimir/apollo-client';
 
 const HistoryOfDonatePage = () => {
-  return <div>History of donate</div>;
+  const { t } = useTranslation();
+  const { id } = useAppSelector((state) => state.user);
+  const { data, loading } = useGetAllDonatedMaterialsByPersonQuery({
+    variables: { id: id.toString() },
+  });
+  return (
+    <Wrapper>
+      <TextArticle>{t('DonateHistory.Title')}</TextArticle>
+      <TextBase>{t('DonateHistory.Desc')}</TextBase>
+      <List>
+        {!loading &&
```

Comment:
if im not mistaken, gql query can return error as well and `loading` will be present, so better to check not for `!loading` but rather for `data`

Reply:
`data` will be null. And if we want to catch an error we need to add `onError` in destruction